Handle missing fine location permission for Android 12

In Android 12 users have the system choice to grant coarse location permission only when asked for during c:geo installation.

This situation seems not to be handled by us, see stacktrace from API31 emulator:

```
2021-11-21 22:01:20.795 6643-6643/cgeo.geocaching E/AndroidRuntime: FATAL EXCEPTION: main
    Process: cgeo.geocaching, PID: 6643
    io.reactivex.rxjava3.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.SecurityException: Neither user 10148 nor current process has android.permission.ACCESS_FINE_LOCATION.
        at io.reactivex.rxjava3.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:718)
        at io.reactivex.rxjava3.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:715)
        at io.reactivex.rxjava3.internal.observers.LambdaObserver.onError(LambdaObserver.java:77)
        at io.reactivex.rxjava3.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.checkTerminated(ObservableObserveOn.java:281)
        at io.reactivex.rxjava3.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:172)
        at io.reactivex.rxjava3.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:255)
        at io.reactivex.rxjava3.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:123)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7839)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
     Caused by: java.lang.SecurityException: Neither user 10148 nor current process has android.permission.ACCESS_FINE_LOCATION.
        at android.os.Parcel.createExceptionOrNull(Parcel.java:2425)
        at android.os.Parcel.createException(Parcel.java:2409)
        at android.os.Parcel.readException(Parcel.java:2392)
        at android.os.Parcel.readException(Parcel.java:2334)
        at android.location.ILocationManager$Stub$Proxy.registerGnssStatusCallback(ILocationManager.java:1876)
        at android.location.LocationManager$GnssStatusTransportManager.registerTransport(LocationManager.java:2990)
        at android.location.LocationManager$GnssStatusTransportManager.registerTransport(LocationManager.java:2980)
        at com.android.internal.listeners.ListenerTransportManager.addListener(ListenerTransportManager.java:70)
        at android.location.LocationManager.registerGnssStatusCallback(LocationManager.java:2431)
        at android.location.LocationManager.registerGnssStatusCallback(LocationManager.java:2412)
        at android.location.LocationManager.registerGnssStatusCallback(LocationManager.java:2389)
        at cgeo.geocaching.sensors.GnssStatusProvider.lambda$createGNSSObservable$1(GnssStatusProvider.java:75)
        at cgeo.geocaching.sensors.-$$Lambda$GnssStatusProvider$Yz9zokRA8lesgjvy_DdM1q5v-uI.subscribe(Unknown Source:2)
        at io.reactivex.rxjava3.internal.operators.observable.ObservableCreate.subscribeActual(ObservableCreate.java:41)
        at io.reactivex.rxjava3.core.Observable.subscribe(Observable.java:13131)
        at io.reactivex.rxjava3.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
        at io.reactivex.rxjava3.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:123)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.os.HandlerThread.run(HandlerThread.java:67)
     Caused by: android.os.RemoteException: Remote stack trace:
        at android.app.ContextImpl.enforce(ContextImpl.java:2178)
        at android.app.ContextImpl.enforceCallingOrSelfPermission(ContextImpl.java:2206)
        at com.android.server.location.gnss.GnssManagerService.registerGnssStatusCallback(GnssManagerService.java:158)
2021-11-21 22:01:20.795 6643-6643/cgeo.geocaching E/AndroidRuntime:     at com.android.server.location.LocationManagerService.registerGnssStatusCallback(LocationManagerService.java:1002)
        at android.location.ILocationManager$Stub.onTransact(ILocationManager.java:830)
```

To reproduce:
- Install c:geo on API31 (Android 12) emulator
- During wizard select to allow coarse location only
- Complete wizward
- Close c:geo and try to restart it

--- a/main/src/cgeo/geocaching/permission/PermissionHandler.java
+++ b/main/src/cgeo/geocaching/permission/PermissionHandler.java
@@ -23,11 +23,10 @@ public class PermissionHandler {
     }
 
     public static void executeIfLocationPermissionGranted(final Activity activity, final PermissionGrantedCallback callback) {
-        final String[] locationPermissions = {Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION};
+        final String[] locationPermissions = {Manifest.permission.ACCESS_FINE_LOCATION};
         final PermissionKey pk = new PermissionKey(locationPermissions);
 
-        if (ContextCompat.checkSelfPermission(activity, locationPermissions[0]) != PackageManager.PERMISSION_GRANTED
-                && ContextCompat.checkSelfPermission(activity, locationPermissions[1]) != PackageManager.PERMISSION_GRANTED) {
+        if (ContextCompat.checkSelfPermission(activity, locationPermissions[0]) != PackageManager.PERMISSION_GRANTED) {
             if (!callbackRegistry.containsKey(pk)) {
                 callbackRegistry.put(pk, new ArrayList<>());
                 ActivityCompat.requestPermissions(activity, locationPermissions, callback.getRequestCode());
@@ -139,4 +138,3 @@ public class PermissionHandler {
     }
 
 }
-

--- a/main/src/cgeo/geocaching/sensors/GeoData.java
+++ b/main/src/cgeo/geocaching/sensors/GeoData.java
@@ -20,7 +20,6 @@ public class GeoData extends Location {
 
     private static final String INITIAL_PROVIDER = "initial";
     private static final String HOME_PROVIDER = "home";
-    private static final String FUSED_PROVIDER = "fused";
     public static final String LOW_POWER_PROVIDER = "low-power";
 
     // Some devices will not have the last position available (for example the emulator). In this case,
@@ -54,8 +53,7 @@ public class GeoData extends Location {
                 return LocationProviderType.GPS;
             case LocationManager.NETWORK_PROVIDER:
                 return LocationProviderType.NETWORK;
-            case FUSED_PROVIDER:
-                // LocationManager.FUSED_PROVIDER constant is not available at API level 9
+            case LocationManager.FUSED_PROVIDER:
                 return LocationProviderType.FUSED;
             case LOW_POWER_PROVIDER:
                 return LocationProviderType.LOW_POWER;

